reflection.md

```diff
@@ -1,18 +1,75 @@
 # Reflection on implementing regular expressions of a DSL
 
 ## Which operators were easiest to implement and why?
+The binary operators were the easiest to implement.
+It was very easy for me to specify the function names as the symbols for the
+operator and to just call `Union` and `Concat`.
+It is simply a wrapper function with symbols as the function name that 
+constructs the appropriate object.
 
 ## Which operators were most difficult to implement and why?
+The most difficult operator to implement is the repetition operator, where
+it takes in the number of repetitions.
+I was not aware of the `apply` function, so I was confused on how the 
+`RegularExpression` object can take in an argument and how that calls a 
+function.
+
+I had a convoluted plan to use currying to create a function that takes in two
+arguments: an operator and an integer.
+Then I would use implicit conversion to convert the `RegularExpression` object
+such that it can infer that it's the curry function that's been given the
+concatenation operator but not the integer argument.
+
+Thankfully, I talked to Prof. Ben and he told me about the `apply` function.
+Once I was aware of this function, it was fairly straightforward to implement
+the operator.
 
 ## Comment on the design of this internal DSL
+It's easy to create categories of characters using the `||` operator.
+For example, we have `digits` in `Program.scala`.
+Since the English language has a small set of characters, relative to other
+languages such as Mandarin Chinese, we can list all the characters within
+a group.
+This process of listing every character in the category may be tedious,
+but it can be made less tedious by defining a variable which stores this 
+information.
```

Comment:
Most languages have more robust regex support, and support things like `r"[a-z]"`.